src/iinit.cpp

```diff
@@ -66,11 +69,225 @@ mkrodsdir() {
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+        else if (!env_has_value) {
+            std::strncpy(_setting, default_value, _len);
+        }
+    } // set_env_from_prompt
+
+    auto set_env_from_prompt(int& _setting, int _default, const char* _prompt) -> void
+    {
+        const bool env_has_value = 0 != _setting;
+        const auto default_value = env_has_value ? _setting : _default;
+
+        fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), default_value);
+        std::string response;
+        std::getline(std::cin, response);
+
+        // The else case indicates that the response was empty and the environment already has a value. If the
+        // environment has a value already, the default value is the existing value (meaning, there will be no
+        // change in the environment). Therefore, the else case is a no-op as no change will occur.
+        if (!response.empty()) {
+            try {
+                _setting = boost::lexical_cast<int>(response);
+            }
+            catch (const boost::bad_lexical_cast&) {
+                _setting = default_value;
+            }
+        }
+        else if (!env_has_value) {
+            _setting = default_value;
+        }
+    } // set_env_from_prompt
+
+    auto configure_required_settings_in_env(RodsEnvironment& _env, nlohmann::json& _json_env) -> void
+    {
+        if (0 == std::strlen(_env.rodsHost)) {
+            constexpr const char* host_prompt = "Enter the host name (DNS) of the server to connect to";
+            set_env_from_prompt(_env.rodsHost, host_prompt, sizeof(_env.rodsHost));
+            _json_env[irods::KW_CFG_IRODS_HOST] = _env.rodsHost;
+        }
+
+        if (0 == _env.rodsPort) {
+            constexpr const char* port_prompt = "Enter the port number";
+            constexpr int default_port = 1247;
+            set_env_from_prompt(_env.rodsPort, default_port, port_prompt);
+            _json_env[irods::KW_CFG_IRODS_PORT] = _env.rodsPort;
+        }
+
+        if (0 == std::strlen(_env.rodsUserName)) {
+            constexpr const char* username_prompt = "Enter your iRODS user name";
+            set_env_from_prompt(_env.rodsUserName, username_prompt, sizeof(_env.rodsUserName));
+            _json_env[irods::KW_CFG_IRODS_USER_NAME] = _env.rodsUserName;
+        }
+
+        if (0 == std::strlen(_env.rodsZone)) {
+            constexpr const char* zone_prompt = "Enter your iRODS zone";
+            set_env_from_prompt(_env.rodsZone, zone_prompt, sizeof(_env.rodsZone));
+            _json_env[irods::KW_CFG_IRODS_ZONE] = _env.rodsZone;
+        }
+    } // configure_required_settings_in_env
+
+    auto configure_ssl_in_env(RodsEnvironment& _env, nlohmann::json& _json_env) -> void
+    {
+        // If the user indicated that SSL is going to be used, these settings are required, so no prompts occur.
```

Comment:
"these settings are required, so no prompts occur." -> "this setting is required, so no prompt shown."

Reply:
"prompt shown" -> "prompt is shown"

Reply:
Done